### ISSUE TITLE
Requesting .withNgComponent

### ISSUE THREAD

**Issue by plinkpaste** [wobkenh/simplematcomponents#30]
It would be really handy to have support to inject a component into a column, an example like so:

```
columns = [
new TableColumn<Data, 'value'>('Value', 'value').withNgComponent<CustomComponent>(
(data) => return {value1: data.value1, value2: data.value2};
)
```

Where the argument to `withNgComponent `defines the `@Input` parameter values in that component.

There might be a better way of doing that, but it would be really useful.

**Comment by wobkenh** [comment]
Thank you very much for creating this issue.

I toyed with the idea of custom components back when I started working on simplemattable, but I deemed it too hard / complicated back then. I gave it another shot now and I think I was succesfull.

I added the function `withNgComponent` to `TableColumn`. Pass your component name as parameter. If you have input parameters in your component, pass a function to the `withNgComponentInput` function of `TableColumn`. Component creation and change detection will be done by simplemattable.

I've provided an example [in the demo application](https://simplex24.de/smc-demo/custom-components). There, you can find an entry "Custom Components" in the navigation list.

To test the new changes, update simplemattable to version `3.4.0-beta.0`. Feel free to test this version and report back if you find any bugs or unwanted behavior.

**Comment by wobkenh** [comment]
I just noticed I forgot a) to merge in the changes of the last small simplemattable update and b) to add the beta tag while publishing to npm yesterday. Today, I did some more testing and I believe now that existing use cases should not be affected by the custom component feature. I merged the changes into master and published as version `3.5.0`.

**Comment by plinkpaste** [comment]
Wow, that's incredible! 

I'll check this out tomorrow.

Thank you very much!
